### PR TITLE
NAT rules for ExternalCIDR (4/4)

### PR DESCRIPTION
**Pull request by davidefalcone1**
# Description
This PR adds a new controller that reconciles the resource NatMapping. It ensures the appropriate set of NAT rules for supporting ExternalCIDR traffic is in place and updated.

### CODE REVIEW

File: internal/liqonet/tunnel-operator/natmapping_operator_test.go [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+	})
+	AfterEach(func() {
+		err := k8sClient.DeleteAllOf(context.Background(), &v1alpha1.NatMapping{}, &client.DeleteAllOfOptions{
+			ListOptions: client.ListOptions{
+				Namespace: namespace,
+			},
+		})
+		Expect(err).To(BeNil())
+	})
+	Context("If a natmapping resource is deleted", func() {
```
Comment (alacuku) [inline internal/liqonet/tunnel-operator/natmapping_operator_test.go:65]: The operator does not add finalizers to the resources, hence when a resource is deleted the operator just can not fetch it and will receive an `notFoundError`. This test case is the same as the one in the context `Reconcile on a resource that does not exist`

File: internal/liqonet/tunnel-operator/tunnel_operator_suite_test.go [hunk 2]
```diff
@@ -1,17 +1,176 @@
...
+	iptNetns, err = netns.CreateNetns(iptNetnsName)
+	Expect(err).To(BeNil())
+
+	// Config IPTables for remote clusters
+	err = initIPTables()
+	Expect(err).To(BeNil())
+
+	err = netv1alpha1.AddToScheme(scheme.Scheme)
+	Expect(err).To(BeNil())
+	//+kubebuilder:scaffold:scheme
```
Comment (alacuku) [inline internal/liqonet/tunnel-operator/tunnel_operator_suite_test.go:98]: we should remove it.

**Review by palexster — APPROVED** [review]

**Review by alacuku — APPROVED** [review]

**Pull request merged**